modules/apps/login/login-web/src/main/java/com/liferay/login/web/internal/portlet/action/LoginMVCActionCommand.java

```diff
@@ -234,11 +236,31 @@ protected void login(
 
 		if (Validator.isNotNull(redirect)) {
 			if (!themeDisplay.isSignedIn()) {
-				actionRequest.setAttribute(
-					WebKeys.REDIRECT,
-					_http.addParameter(
-						_portal.getPathMain() + "/portal/login", "redirect",
-						redirect));
+				User user = _userLocalService.getUserByEmailAddress(
+					themeDisplay.getCompanyId(),
+					actionRequest.getParameter("login"));
```

Comment:
:raised_eyebrow: The portal can be configured to use 3 different logins: `emailAddress`, `screenName` and `userId` so the `login` param is not necessarily an email address.

Reply:
Do you have a suggestion for how I can get the value of `user.isPasswordReset()` using another method? I spent a lot of time trying to find a solution suitable for both LPS-134303 and LPS-121766 and this was the only fix I could come up with.

Any advice would be appreciated!
Thanks!

Reply:
Hello @noornajj,

I think the problem is with this check:

```
if (!themeDisplay.isSignedIn()) {
```

This does not make sense to me.

The themeDisplay instance we have at that point was initialized before the authentication with `signedIn` set to `false` so it can't return true. It is set here: https://github.com/lipusz/liferay-portal/blob/master/portal-impl/src/com/liferay/portal/events/ServicePreAction.java#L894, and there is no interim ServicePreAction during the authentication so it remains null.

Could you blame the file to see when and why it was added like that?

I did a quick test and manually updated the flag to `true` during debugging to see if I get redirected to the expected private page, and yes, I was.

HTH.

--

Tibor

Reply:
Hello @lipusz,

This was added in [LPS-78502](https://issues.liferay.com/browse/LPS-78502)

It was implemented because the LastLoginDate value wasn't being updated in the database when an authenticated user access a link to a document. The fix has LoginMVCActionCommand redirect to itself first to make sure MainServlet is hit and login dates columns in DB are updated.

Reply:
Hi @lipusz,

Do you have an update for this PR?

Thank you.

Reply:
Hi @noornajj @lipusz, 

Regarding the user retrieval, one option we could try is retrieving the userId from the session attribute: **j_username**
- `session.getAttribute("j_username")`

Reply:
Thank you @ericyanLr  for the suggestion. I will test this and update with the results.

Reply:
This alternative works. @lipusz, is this an appropriate fix?

Reply:
j_username is only set when JAAS login is used, it's not going to work either.

I tend to think that this "regression" should nit be fixed and the customer should be advised to find a different method for redirecting a subset of users (who are not authenticated through SAML) to the private page after the portal login.

Reply:
Hello @lipusz,
Thank you for your response. From the discussion on LPP-41225, I see that there isn't an alternative URL that the customer can use to bypass SAML and redirect to the desired page. The only other method I can think of is to have those users be redirected through a 2 step process. First to login through the URL that bypasses SAML, and then, once they are authenticated, ask them to access another link that redirects them to the private page.
What is your opinion of this method?